tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
OK, so the argument in favour of removing this special casing is that tsfc doesn't know about what constitutes a "small loop" and therefore should not be eliminated.  So it should always annotate zero columns.  Rather than using the heuristic that "if it's a mixed or enriched element then there will probably be *enough* zero columns".

The argument in favour of keeping it is that coffee currently doesn't check how many zero columns it's eliminating, and always does so?